common/concurrency/interval_runner.rs

```diff
@@ -29,9 +29,16 @@ impl IntervalRunner {
     ) -> Self {
         let (shutdown_sender, shutdown_receiver) = sync_channel::<SyncSender<()>>(1);
         thread::spawn(move || {
-            if !initial_delay.is_zero() {
-                thread::sleep(initial_delay);
+            match shutdown_receiver.recv_timeout(initial_delay) {
+                Ok(done_sender) => {
+                    drop(action);
+                    done_sender.send(()).unwrap();
+                    return;
+                }
+                Err(RecvTimeoutError::Timeout) => (),
+                Err(RecvTimeoutError::Disconnected) => return, // TODO log?
             }
```

Comment:
Without this change, `^C` hangs until this delay is up. This is because the `Drop` of `IntervalRunner` waits for acknowledgement from the background thread and will not complete until then.